mmv1/third_party/terraform/services/compute/resource_compute_disk_test.go.tmpl

```diff
@@ -1644,6 +1668,61 @@ resource "google_compute_disk" "foobar" {
 `, diskName, storagePoolUrl)
 }
 
+func TestExpandStoragePoolUrl_withEmptyConfigProjectAndZone(t *testing.T) {
+	config := &transport_tpg.Config{
+		ComputeBasePath: "https://www.googleapis.com/compute/v1/",
+	}
+
+	data := &mockTerraformResourceData{
+		m: map[string]interface{}{
+			"project": "test-project",
+			"zone": "test-zone",
+		},
+	}
```

Comment:
we provide a struct that mocks this interface that you can use here: [tpgresource.ResourceDataMock](https://github.com/hashicorp/terraform-provider-google/blob/d380eca5a49cf0b8ca4b2737a22c6704005c5761/google/tpgresource/resource_test_utils.go#L18)

```suggestion
	data := &tpgresource.ResourceDataMock{
		FieldsInSchema: map[string]interface{}{
			"project": "test-project",
			"zone": "test-zone",
		},
	}
```